In [12]:
import pandas as pd
import numpy as np

In [13]:
input_data_columns = ['Sentence_1', 'Sentence_2', 'Source_1', 'Source_2']
input_data = pd.read_csv('sts2016-english-with-gs-v1.0/STS2016.input.answer-answer.txt', sep='\t', header=None, names=input_data_columns)

In [14]:
data = []
with open('sts2016-english-with-gs-v1.0/STS2016.gs.answer-answer.txt', 'r') as file:
    for line in file:
        # If the line is empty or just contains a newline character, add NaN
        if line.strip() == '':
            data.append([np.nan])
        else:
            # Otherwise, add the value to the data list
            data.append([line.strip()])
    

In [15]:
data_label_df = pd.DataFrame(data)
data_label_df = pd.DataFrame(data, columns=['Value'])
print(len(data_label_df))

1571


Da wir 3 Zeilen zu viel haben, untersuchen wir diese. Wenn diese NaN sind, löschen wir diese.

In [16]:
for i in range(1, 5):
    if pd.isna(data_label_df.iloc[-i]).any():  # Überprüfe, ob alle Werte in der Zeile NaN sind
        data_label_df = data_label_df.drop(data_label_df.index[-i])

In [17]:
print(data_label_df.info())

<class 'pandas.core.frame.DataFrame'>
Index: 1567 entries, 0 to 1569
Data columns (total 1 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Value   254 non-null    object
dtypes: object(1)
memory usage: 24.5+ KB
None


In [18]:
# print(label.describe())

In [19]:
print(input_data.head())

                                          Sentence_1  \
0               Tasting it is the only reliable way.   
1         I think it probably depends on your money.   
2  You need to read a lot to know what you like a...   
3  Obviously, the best book for you depends a lot...   
4                        I've had this same problem.   

                                          Sentence_2  \
0                       The way you have it is fine.   
1                        It depends on your country.   
2                            You don't have to know.   
3  The answer will depend of course on what you'r...   
4                     I had the same problem as you.   

                                            Source_1  \
0  StackExchange Network: http://cooking.stackexc...   
1  StackExchange Network: http://workplace.stacke...   
2  StackExchange Network: http://writers.stackexc...   
3  StackExchange Network: http://travel.stackexch...   
4  StackExchange Network: http://diy.stackexch

In [20]:
print(input_data.describe())

                                               Sentence_1  \
count                                                1571   
unique                                                532   
top     You need to read a lot to know what you like a...   
freq                                                   38   

                         Sentence_2  \
count                          1571   
unique                         1189   
top     It depends on your country.   
freq                             13   

                                                 Source_1  \
count                                                1571   
unique                                                532   
top     StackExchange Network: http://writers.stackexc...   
freq                                                   38   

                                                 Source_2  
count                                                1571  
unique                                               1189  
top     Stac

In [21]:
count_same_elements_sentences = (input_data.Sentence_1 == input_data.Sentence_2).sum()
count_same_elements_sources = (input_data.Source_1 == input_data.Sentence_2).sum()

In [22]:
print(count_same_elements_sentences)
print(count_same_elements_sources)

0
0


In [23]:
freq_sentence1 = input_data.Sentence_1.value_counts()
freq_sentence2 = input_data.Sentence_2.value_counts()

freq_source1 = input_data.Source_1.value_counts()
freq_source2 = input_data.Source_2.value_counts()

In [24]:
print(freq_sentence1)
print(freq_sentence2)

Sentence_1
You need to read a lot to know what you like and what you don't.                  38
It depends on what you want to do next, and where you want to do it.              34
You just have to base your answer on what you do know, which is what you want.    30
You don't have to know.                                                           28
In the first case, I think you don't need it.                                     23
                                                                                  ..
I think you can get away with thinset here.                                        1
the best thing to do with toddlers when it comes to food is to stop caring.        1
Interesting question but ultimately the answer is no.                              1
I'm a huge fan of the vintage glass bowl/reamer combination.                       1
Voltron is the name of the show you're looking for                                 1
Name: count, Length: 532, dtype: int64
Sentence_2
It d

In [25]:
print(freq_source1)

Source_1
StackExchange Network: http://writers.stackexchange.com/questions/12166 Author: Lexi (http://writers.stackexchange.com/users/2410)                                                                                       38
StackExchange Network: http://academia.stackexchange.com/questions/26950 Author: Dylan Richard Muir (http://academia.stackexchange.com/users/19984)                                                                      34
StackExchange Network: http://workplace.stackexchange.com/questions/9526 Author: pdr (http://workplace.stackexchange.com/users/759) Last Editor: pdr (http://workplace.stackexchange.com/users/759)                      30
StackExchange Network: http://academia.stackexchange.com/questions/26241 Author: BigPants (http://academia.stackexchange.com/users/19703) Last Editor: BigPants (http://academia.stackexchange.com/users/19703)          28
StackExchange Network: http://writers.stackexchange.com/questions/10368 Author: Driss Zouak (http://writers.sta

In [26]:
print(freq_source2)

Source_2
StackExchange Network: http://travel.stackexchange.com/questions/45030 Author: Vince (http://travel.stackexchange.com/users/3044) Last Editor: Vince (http://travel.stackexchange.com/users/3044)                                    13
StackExchange Network: http://academia.stackexchange.com/questions/26241 Author: BigPants (http://academia.stackexchange.com/users/19703) Last Editor: BigPants (http://academia.stackexchange.com/users/19703)                       9
StackExchange Network: http://academia.stackexchange.com/questions/27341 Author: yo' (http://academia.stackexchange.com/users/1471)                                                                                                   9
StackExchange Network: http://pets.stackexchange.com/questions/6824 Author: Delusion (http://pets.stackexchange.com/users/3783)                                                                                                       7
StackExchange Network: http://diy.stackexchange.com/questions/6

Wir können sehen, dass es insgesamt pro Spalte (Sentence_1-2), insgesamt 1568 Sätze gibt.
Für die erste Spalte gibt es 532 verschiedene Sätze, und für Spalte 2 533 verschiedene Sätze.
Insgesamt sind die Spalten aber Disjunkt, dass heißt, ein Satz in der ersten Spalte kommt NICHT in der zweiten Spalte vor.

Der am meisten vorkommende Satz in Spalte 1 ist: "You need to read a lot to know what you like and what you don't." mit Häufigkeit 38 und in Spalte 2 der Satz: "It depends on your country." mit Häufigkeit 13.

Ebenfalls sind die Source-Spalten Disjunkt, mit ebenfalls 1568 Einträgen.
Für die erste Source-Spalte haben wir 533 diverse, und für die zweite Spalte 1190 diverse Einträge.

Die am meisten auftauchende Quelle in Spalte 1 ist (mit Häufigkeit 38): "http://writers.stackexchange.com/questions/12166"
Die am meisten auftauchende Quelle in Spalte 2 ist (mit Häfugkeit 13):  "http://travel.stackexchange.com/questions/45030"